### PR TITLE
AppVeyor CI Integration 2 (monorepo update)

### PR DESCRIPTION
**Pull request by patkub**
Update to where I left off with #80. This now works and runs directly on AppVeyor (no need for bash). I added an option to ci-services to change the way command output is ignored to make it compatible.

Pull Request where greenkeeper updated the lockfile:
https://github.com/patkub/test-greenkeeper-lockfile-appveyor/pull/22

Successful branch build where greenkeeper-lockfile updated, committed, and pushed the new lockfile:
https://ci.appveyor.com/project/patkub/test-greenkeeper-lockfile-appveyor/build/1.0.259

Successful build that ran on the PR:
https://ci.appveyor.com/project/patkub/test-greenkeeper-lockfile-appveyor/build/1.0.260

Successful branch build on "chore(package): update lockfile" commit:
https://ci.appveyor.com/project/patkub/test-greenkeeper-lockfile-appveyor/build/1.0.261

Successful build that ran on the PR:
https://ci.appveyor.com/project/patkub/test-greenkeeper-lockfile-appveyor/build/1.0.262

AppVeyor config: https://github.com/patkub/test-greenkeeper-lockfile-appveyor/blob/master/appveyor.yml

Travis jest tests are now passing with 100% coverage.

### CODE REVIEW

**Comment by patkub** [comment]
Could I somehow ignore running on "chore(package): update lockfile" commits?

File: lib/update-lockfile.js [hunk 1]
```diff
@@ -52,17 +53,18 @@ module.exports = function updateLockfile (dependency, options) {
   if (exec('git status --porcelain').toString() === '') return
 
   // stage the updated lockfile
-  exec('git add npm-shrinkwrap.json 2>/dev/null || true')
-  exec('git add package-lock.json 2>/dev/null || true')
-  exec('git add yarn.lock 2>/dev/null || true')
+  const ignoreOutput = info.ignoreOutput ? info.ignoreOutput : '2>/dev/null || true'
+  exec(`git add npm-shrinkwrap.json ${ignoreOutput}`)
+  exec(`git add package-lock.json ${ignoreOutput}`)
+  exec(`git add yarn.lock ${ignoreOutput}`)
```
Comment (patkub) [inline lib/update-lockfile.js]: Here I'm using the method for ignoring output defined in the ci-service, if its available.

File: lib/update-lockfile.js [hunk 2]
```diff
@@ -52,17 +53,18 @@ module.exports = function updateLockfile (dependency, options) {
...
   exec(`git config user.email "${commitEmail}"`)
   exec(`git config user.name "${commitName}"`)
 
   if (shouldAmend) {
     exec(`git commit --amend --author="${commitName} <${commitEmail}>" --no-edit`)
   } else {
-    const updateMessage = 'chore(package): update lockfile\n\nhttps://npm.im/greenkeeper-lockfile'
-    exec(`git commit -m "${updateMessage}"`)
+    const updateMessage = info.updateMessage ? info.updateMessage : '-m "chore(package): update lockfile\n\nhttps://npm.im/greenkeeper-lockfile"'
+    exec(`git commit ${updateMessage}`)
```
Comment (patkub) [inline lib/update-lockfile.js]: Here I'm using the update message defined in the ci-service, if its available.

File: lib/update-lockfile.js [hunk 3]
```diff
@@ -53,7 +52,7 @@ module.exports = function updateLockfile (dependency, options) {
   if (exec('git status --porcelain').toString() === '') return
 
   // stage the updated lockfile
-  const ignoreOutput = info.ignoreOutput ? info.ignoreOutput : '2>/dev/null || true'
+  const ignoreOutput = options.ignoreOutput ? options.ignoreOutput : '2>/dev/null || true'
```
Comment (patkub) [inline lib/update-lockfile.js]: Here I'm using the method for ignoring output defined in the ci-service, if its available.

File: lib/update-lockfile.js [hunk 4]
```diff
@@ -64,7 +63,7 @@ module.exports = function updateLockfile (dependency, options) {
   if (shouldAmend) {
     exec(`git commit --amend --author="${commitName} <${commitEmail}>" --no-edit`)
   } else {
-    const updateMessage = info.updateMessage ? info.updateMessage : '-m "chore(package): update lockfile\n\nhttps://npm.im/greenkeeper-lockfile"'
+    const updateMessage = options.updateMessage ? options.updateMessage : '-m "chore(package): update lockfile\n\nhttps://npm.im/greenkeeper-lockfile"'
```
Comment (patkub) [inline lib/update-lockfile.js]: Here I'm using the update message defined in the ci-service, if its available.

File: ci-services/appveyor.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  repoSlug: env.APPVEYOR_REPO_NAME,
+  branchName: env.APPVEYOR_REPO_BRANCH,
+  firstPush: shouldUpdate(),
+  correctBuild: !env.APPVEYOR_PULL_REQUEST_NUMBER,
+  uploadBuild: true,
+  // Used to ignore output when staging the updated lockfile
+  // The Windows equivalent of '2>/dev/null || true' is '>NUL || (exit 0)'
+  ignoreOutput: '>NUL || (exit 0)',
+  // Multi-line update message because committing with linebreaks does not work.
+  updateMessage: '-m "chore(package): update lockfile" -m "https://npm.im/greenkeeper-lockfile"'
```
Comment (patkub) [inline ci-services/appveyor.js]: Changing updateMessage is necessary. New line characters \n\n don't work in AppVeyor. Here is a failed build when this option is not set: https://ci.appveyor.com/project/patkub/test/build/1.0.5

File: ci-services/appveyor.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  repoSlug: env.APPVEYOR_REPO_NAME,
+  branchName: env.APPVEYOR_REPO_BRANCH,
+  firstPush: shouldUpdate(),
+  correctBuild: !env.APPVEYOR_PULL_REQUEST_NUMBER,
+  uploadBuild: true,
+  // Used to ignore output when staging the updated lockfile
+  // The Windows equivalent of '2>/dev/null || true' is '2>NUL || (exit 0)'
+  ignoreOutput: '2>NUL || (exit 0)',
+  // Multi-line update message because committing with linebreaks does not work.
+  updateMessage: '-m "chore(package): update lockfile" -m "https://npm.im/greenkeeper-lockfile"'
```
Comment (patkub) [inline ci-services/appveyor.js]: Changing updateMessage is necessary. New line characters \n\n don't work in AppVeyor. Here is a failed build when this option is not set: https://ci.appveyor.com/project/patkub/test/build/1.0.5
https://github.com/patkub/greenkeeper-lockfile/tree/appveyor-ci-service-no-update-message

File: lib/update-lockfile.js [hunk 7]
```diff
@@ -52,17 +52,18 @@ module.exports = function updateLockfile (dependency, options) {
...
   exec(`git config user.email "${commitEmail}"`)
   exec(`git config user.name "${commitName}"`)
 
   if (shouldAmend) {
     exec(`git commit --amend --author="${commitName} <${commitEmail}>" --no-edit`)
   } else {
-    const updateMessage = 'chore(package): update lockfile\n\nhttps://npm.im/greenkeeper-lockfile'
-    exec(`git commit -m "${updateMessage}"`)
+    const updateMessage = options.updateMessage ? options.updateMessage : '-m "chore(package): update lockfile\n\nhttps://npm.im/greenkeeper-lockfile"'
+    exec(`git commit ${updateMessage}`)
```
Comment (patkub) [inline lib/update-lockfile.js]: Here I'm using the update message defined in the ci-service, if its available.

File: lib/update-lockfile.js [hunk 8]
```diff
@@ -52,17 +52,18 @@ module.exports = function updateLockfile (dependency, options) {
   if (exec('git status --porcelain').toString() === '') return
 
   // stage the updated lockfile
-  exec('git add npm-shrinkwrap.json 2>/dev/null || true')
-  exec('git add package-lock.json 2>/dev/null || true')
-  exec('git add yarn.lock 2>/dev/null || true')
+  const ignoreOutput = options.ignoreOutput ? options.ignoreOutput : '2>/dev/null || true'
+  exec(`git add npm-shrinkwrap.json ${ignoreOutput}`)
+  exec(`git add package-lock.json ${ignoreOutput}`)
+  exec(`git add yarn.lock ${ignoreOutput}`)
```
Comment (patkub) [inline lib/update-lockfile.js]: Here I'm using the ignore output method defined in the ci-service, if its available.

**Comment by patkub** [comment]
It is only necessary to change the way the output is ignored, not the syntax of the commit message.

**Comment by patkub** [comment]
This now works with monorepo support, but the jest tests are still broken.
https://github.com/patkub/test-greenkeeper-lockfile-appveyor/compare/greenkeeper/test-greenkeeper-lockfile-appveyor-package-1.0.69

**Comment by Realtin** [comment]
@patkub can you rebase this?

**Pull request closed without merging**